File: cpp/13.cpp
class Solution {
 public:
  int romanToInt(string s) {
    int r2i[256], ret = 0, last = 'I';
    r2i['I'] = 1;
    r2i['V'] = 5;
    r2i['X'] = 10;
    r2i['L'] = 50;
    r2i['C'] = 100;
    r2i['D'] = 500;
    r2i['M'] = 1000;
    for (int i = s.size() - 1; i >= 0; --i) {
      if (r2i[s[i]] < r2i[last])
        ret -= r2i[s[i]];
      else {
        ret += r2i[s[i]];
        last = s[i];
      }
    }
    return ret;
  }
};

File: js/13.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function (s) {
  let r2i = {
      I: 1,
      V: 5,
      X: 10,
      L: 50,
      C: 100,
      D: 500,
      M: 1000
    },
    ret = 0,
    last = 'I'
  for (let i = s.length - 1; i >= 0; --i) {
    if (r2i[s[i]] < r2i[last]) ret -= r2i[s[i]]
    else ret += r2i[s[i]], last = s[i]
  }
  return ret
};